### PR TITLE
Create NotificationModalViewController

### PR DESCRIPTION
**Pull request by ReadeP**
## Overview

- Create `NotificationModalViewController`
- Fix constraint issues with iPhone SE 🙌

## Changes Made

- Create `NotificationModalViewController`
- Add temporary support for testing using a shake gesture
- Fixed constrain issues in `SearchTableViewCell` and `SearchDetailTableViewCell`

## Test Coverage

Tested manually using shake gesture (until we set up push notifications)

## Screenshots

<details>

  <summary>Example</summary>

  
Simulator Screen Shot - iPhone 11 - 2020-02-20 at 11 50 01

  

</details>

## Related Issues

#21 and #22.

### CODE REVIEW

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 1]
```diff
@@ -93,6 +93,14 @@ class MainNavigationController: UINavigationController {
         // Add insets to the back button so it perfectly aligns with the HomeTableViewController back button.
         navigationBar.backIndicatorImage = UIImage.backIcon?.with(insets: UIEdgeInsets(top: 0, left: 8, bottom: 2.5, right: 0))
         navigationBar.backIndicatorTransitionMaskImage = .backIcon
+
+        becomeFirstResponder()
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: Why is this here? I don't think it belongs in this class.

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 2]
```diff
@@ -93,6 +93,14 @@ class MainNavigationController: UINavigationController {
...
+    }
+
+    override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: This is just for testing, right?

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+class NotificationModalViewController: UIViewController {
+
+    private let availableLabel = UILabel()
+    private let courseIDLabel = UILabel()
+    private let courseIDTitleLabel = UILabel()
+    private let courseTitleLabel = UILabel()
+    private let homeButton = UIButton()
+    private let sectionLabel = UILabel()
+    private let courseIDStackView = UIStackView()
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: Alphabetize 👹

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        studentCenterButton.layer.borderWidth = 1
+        studentCenterButton.layer.borderColor = UIColor.white.cgColor
+        studentCenterButton.setTitle("OPEN STUDENT CENTER", for: .normal)
+        studentCenterButton.titleLabel?.font = ._18Semibold
+        studentCenterButton.setTitleColor(.white, for: .normal)
+        studentCenterButton.on(.touchUpInside, openStudentCenter)
+        view.addSubview(studentCenterButton)
+
+        // Setup constraints
+
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: delete newline

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        availableLabel.snp.makeConstraints { make in
+            let constaint = UIScreen.main.nativeBounds.height == 1136 ? 40 : 91
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: did u mean `constant` or `constraint`?
also is `UIScreen.main.nativeBounds.height == 1136` to check if the user is on `SE`? if so can we extract this into a helper function since you seem to use it in multiple places

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            make.top.equalToSuperview().inset(constaint)
+            make.leading.trailing.equalToSuperview().inset(91)
+        }
+
+        courseTitleLabel.snp.makeConstraints { make in
+            let constaint = UIScreen.main.nativeBounds.height == 1136 ? 24 : 43
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: same here haha

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            make.top.equalTo(courseTitleLabel.snp.bottom).offset(24)
+            make.leading.trailing.equalToSuperview().inset(20)
+        }
+
+        courseIDStackView.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+
+        homeButton.snp.makeConstraints { make in
+            let constaint = UIScreen.main.nativeBounds.height == 1136 ? 50 : 145
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: same here

File: CourseGrab/ViewControllers/Login/LoginViewController.swift [hunk 8]
```diff
@@ -56,5 +56,9 @@ class LoginViewController: UIViewController {
         super.viewWillAppear(animated)
         navigationController?.setNavigationBarHidden(true, animated: false)
     }
-    
+
+    override func viewDidAppear(_ animated: Bool) {
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/Login/LoginViewController.swift]: Make sure to call `super.viewDidAppear`

File: CourseGrab/ViewControllers/Login/LoginViewController.swift [hunk 9]
```diff
@@ -56,5 +56,9 @@ class LoginViewController: UIViewController {
...
+        GIDSignIn.sharedInstance()?.restorePreviousSignIn()
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/Login/LoginViewController.swift]: Delete this line. This is automatically handled by Firebase.

File: CourseGrab/ViewControllers/MainNavigationController.swift [hunk 10]
```diff
@@ -95,4 +95,10 @@ class MainNavigationController: UINavigationController {
         navigationBar.backIndicatorTransitionMaskImage = .backIcon
     }
 
+    override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
+        if motion == .motionShake {
+            pushViewController(NotificationModalViewController(), animated: true)
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/MainNavigationController.swift]: Shouldn’t we delete this if were done testing?

File: CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+        section: "DIS 005 / TR 1:25PM",
+        status: .open,
+        subjectCode: "CS",
+        title: "CS 2112: Data Structures and Algorithms, a Very Long Title"
+    )
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        navigationController?.setNavigationBarHidden(true, animated: true)
```
Comment (danielVebman) [inline CourseGrab/ViewControllers/Notification/NotificationModalViewController.swift]: Maybe move under `// Setup appearance` because this is technically appearance?

**Comment by danielVebman** [comment]
Also: add related issues to your PR summary just so the connections are clear. i.e. #21 and #22.

**Comment by danielVebman** [comment]
Bump on deleting the redundant Google Sign in
